cmake/OpenCensusDeps.cmake

```diff
@@ -58,3 +58,25 @@ if(NOT TARGET absl::base)
   add_subdirectory(${CMAKE_BINARY_DIR}/abseil-src
                    ${CMAKE_BINARY_DIR}/abseil-build EXCLUDE_FROM_ALL)
 endif()
+
+configure_file(${CMAKE_CURRENT_SOURCE_DIR}/cmake/prometheus-cpp.CMakeLists.txt
+               ${CMAKE_BINARY_DIR}/prometheus-download/CMakeLists.txt)
+execute_process(COMMAND ${CMAKE_COMMAND} -G "${CMAKE_GENERATOR}" .
+                RESULT_VARIABLE result
+                WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/prometheus-download)
+if(result)
+  message(FATAL_ERROR "CMake step failed: ${result}")
+endif()
+execute_process(COMMAND ${CMAKE_COMMAND} --build .
+                RESULT_VARIABLE result
+                WORKING_DIRECTORY ${CMAKE_BINARY_DIR}/prometheus-download)
+if(result)
+  message(FATAL_ERROR "Build step failed: ${result}")
+endif()
+
+set(ENABLE_PULL OFF CACHE BOOL "Build prometheus-cpp pull library" FORCE)
```

Comment:
Also we need `pull` specifically for `exposer.h`

Reply:
Let's leave this for a followup PR.

Reply:
not adding pull library now because `prometheus::pull` is a library for binaries like `prometheus_test_server` & `helloworld`. For some reason, I cannot add build rule for binary `prometheus_test_server` successfully. so make this PR self contained, I decided to leave the pull out from this PR.

Will add `prometheus-cpp:pull` library back for next PR; will also add guard for `prometheus-cpp:pull` library as well.